fw/http_limits.c

```diff
@@ -1140,14 +1140,22 @@ frang_http_req_process(FrangAcc *ra, TfwConn *conn, TfwFsmData *data,
 
 	/* Ensure that HTTP request method is one of those defined by a user. */
 	T_FSM_STATE(Frang_Req_Hdr_Method) {
-		if (f_cfg->http_methods_mask) {
-			if (req->method == _TFW_HTTP_METH_NONE ||
-			    req->method == _TFW_HTTP_METH_INCOMPLETE) {
-				T_FSM_EXIT();
-			}
-			r = frang_http_methods(req, ra,
-					       f_cfg->http_methods_mask);
+		if (req->method == _TFW_HTTP_METH_NONE ||
+		    req->method == _TFW_HTTP_METH_INCOMPLETE)
+		{
+			T_FSM_EXIT();
 		}
+
+		if (!f_cfg->http_methods_mask) {
+			/* If we have a default value,
+			 * we set all bits to 1 except DELETE and PUT*/
+			f_cfg->http_methods_mask = ~((1 << TFW_HTTP_METH_DELETE) |
+						     (1 << TFW_HTTP_METH_PUT));
```

Comment:
1. I suggest doing it in tfw_frang_clean(FrangVhostCfg *cfg). We set it as default value and if we have something in the config file, it will be changed.

2. What do I have to do with this config?
    frang_limits {
        http_methods ;
    }
Because it also allows everything

3. What about OPTIONS in default?

Reply:
It seems `tfw_vhost_cfgstart()` is a better function to store default values.

Also please check configuration reload (i.e. we had some values for `http_methods` and after that going to use defaults) on-the fly